src/util.c

```diff
@@ -690,7 +690,7 @@ ndsctl_json_all(FILE *fp)
 
 	LOCK_CLIENT_LIST();
 
-	fprintf(fp, "{\n\"client_length\": %d,\n", get_client_list_length());
+	fprintf(fp, "{\n\"client_list_length\": \"%d\",\n", get_client_list_length());
```

Comment:
why do all numbers have to be strings? This seems to be artificial.

Reply:
It makes parsing all the client variables sent by ndsctl very simple. The login.sh (aka demo-preauth.sh) has a very simple parser that extracts all the variables from the json output.